### ISSUE TITLE
Important Tip for Greatly Improving Cinema4D Joint Deforming

### ISSUE THREAD

**Issue by brucek5** [daz3d/DazToC4D#43]
I can't believe I missed this! In Cinema4D, the joint deformer Weight Tag works based on the settings of the Skin tag. By default the Skin tag deforming "Type" is set to "Linear" due to its the fastest to calculate. Regardless of speed, if the "Type" is changed to "Spherical" it makes a huge difference. Once changed to "Spherical", joint deformation will almost look like JCM/ECR have been applied. Mesh kinks and bends in forearms, knees, etc. go away, and like name implies you get a more Spherical deformation rather than a bending deformation.

Bruce

**Comment by brucek5** [comment]
For even more flexibility the Skin Tag type "Blend" enables better control to remove some of the extreme deformations that "Spherical" produces. In Blend type, you can set 0-100% which is all Linear at 0% and full Spherical at 100%, 50% is a mixture of both. You can also supply a Vertex map to the Bend type to define the blend strength differently across the entire mesh.

**Comment by danielbui78** [comment]
Yes, thanks for this tip.  I am actually aware of it but have not found c4d python hooks to enable this in the plugin.  If you find the python instructions to do this programmatically, please let me know.  Thanks!

**Comment by brucek5** [comment]
Hi danielbui78,

I don't know how to embed code nicely in GitHub. Not sure why getting the skin object is so different that getting other objects, but with a lot digging I found how. Attached Python code sets the skin object type and blend mix amount.

[SetSkinObjectSettings.txt](https://github.com/daz3d/DazToC4D/files/9636937/SetSkinObjectSettings.txt)

I suggest you use the "Blend" type with a mix value of at least .5 (50%), but not over .75 (75%). After doing tests on multiple joints, the type "Spherical" can be too aggressive. In my case, I'm going to use the vertex map option to set the mix values for each joint.

I'm glad I was able to contribute to this project,

Bruce.
